server/rpc.go

```diff
@@ -255,23 +258,26 @@ func getOrchestrator(orch Orchestrator, req *net.OrchestratorRequest) (*net.Orch
 		return nil, fmt.Errorf("Invalid orchestrator request (%v)", err)
 	}
 
-	webhookRes, err := authenticateBroadcaster(addr.Hex())
-	if err != nil {
+	if _, err := authenticateBroadcaster(addr); err != nil {
 		return nil, fmt.Errorf("authentication failed: %v", err)
 	}
 
 	// currently, orchestrator == transcoder
-	oInfo, err := orchestratorInfo(orch, addr, orch.ServiceURI().String())
-	if err != nil {
-		return nil, err
-	}
+	return orchestratorInfo(orch, addr, orch.ServiceURI().String())
+}
 
-	// Adjust the necessary orchestrator info values
-	return handleWebhookResponse(webhookRes, oInfo), nil
+func getPriceInfo(orch Orchestrator, addr ethcommon.Address) (*net.PriceInfo, error) {
+	if AuthWebhookURL != "" {
+		webhookRes := getFromDiscoveryAuthWebhookCache(addr)
+		if webhookRes != nil && webhookRes.PriceInfo != nil {
```

Comment:
Will this result in `orch.PriceInfo()` being called if the webhook does not return a price?

How do we expect the webhook server to tell the orchestrator to use its default price? Should the webhook server omit the price field, set the price field to null, or would any of those options suffice? Would be good to document the requirements here.

Reply:
Should definitely document this , we will always call `orch.PrinceInfo` if the webhook response in the cache doesn't have a `priceInfo` field defined or its value is undefined.